core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -702,7 +719,15 @@ private static AnnotatedValueResolver ofInjectableTypes0(AnnotatedElement annota
 
                 // Do not convert value here because the element type is String.
                 if (values != null && !values.isEmpty()) {
-                    values.stream().map(resolver::convert).forEach(resolvedValues::add);
+                    if (queryDelimiter != null && values.size() == 1) {
+                        final String first = values.get(0);
+                        Splitter.on(queryDelimiter)
```

Comment:
Late nit: Maybe we can have `Splitter` as a closure variable so that it doesn't get a new instance on every request?

Reply:
@ghkim3221 What do you think?